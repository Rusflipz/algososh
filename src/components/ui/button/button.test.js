import { Button } from "./button";
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor, screen } from "@testing-library/react"

const buttonText = "text"

describe("Проверка кнопки", () => {

    it("Проверка кнопки с текстом", () => {
        const snap = renderer.create(<Button text="test text" />).toJSON()
        expect(snap).toMatchSnapshot()
    });

    it("Проверка кнопки без текстом", () => {
        const snap = renderer.create(<Button />).toJSON()
        expect(snap).toMatchSnapshot()
    });


    it("Проврка неактивной кнопки", () => {
        const snap = renderer.create(<Button disabled={true} />).toJSON()
        expect(snap).toMatchSnapshot()
    });

    it("Проврка кнопкиr загрузки", () => {
        const snap = renderer.create(<Button isLoader={true} />).toJSON()
        expect(snap).toMatchSnapshot()
    });

    it("Проверка клика", () => {
        const fn = jest.fn()
        render(<Button text={buttonText} onClick={fn} />)
        const button = screen.getByText(buttonText)
        fireEvent.click(button)
    });

})
