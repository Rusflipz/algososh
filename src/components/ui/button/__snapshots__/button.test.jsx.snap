// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render button Checking button disabled render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check render button Checking button disabled render 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Check render button Checking button with text render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;

exports[`Check render button Checking button without text render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверка кнопки Checking button disabled render 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Проверка кнопки Проверка кнопки без текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Проверка кнопки Проверка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;

exports[`Проверка кнопки Проврка кнопкиr загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Проверка кнопки Проврка неактивной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
